trackio/__init__.py

```diff
@@ -34,6 +34,8 @@ def init(
     project: str,
     name: str | None = None,
     space_id: str | None = None,
+    persistent_dataset: str | None = None,
```

Comment:
Can we add docstrings for this param? And for consistency with `space_id`, perhaps we call this:
```suggestion
    dataset_id: str | None = None,
```
the persistence can be explained in the docstring